### PR TITLE
Pyarrow vulnerability fix

### PR DESCRIPTION
**Pull request by hawestra**
use dockerfile instead of conda file for pyarrow

### CODE REVIEW

File: assets/responsibleai/environments/responsibleai-ubuntu20.04-py38-cpu/context/conda_dependencies.yaml [hunk 1]
```diff
@@ -9,8 +9,6 @@ dependencies:
   - pip:
     - responsibleai~=0.30.0
     - raiwidgets~=0.30.0
-    - pyarrow
-    - pyarrow-hotfix>=0.5
```
Comment (gaugup) [inline assets/responsibleai/environments/responsibleai-ubuntu20.04-py38-cpu/context/conda_dependencies.yaml:13]: We don't need pyarrow-hotfix?

Comment (hawestra) [reply assets/responsibleai/environments/responsibleai-ubuntu20.04-py38-cpu/context/conda_dependencies.yaml:13]: no, the only reason we need it is if we are using pyarrow with the vulnerability https://github.com/advisories/GHSA-5wvp-7f3h-6wmm: 
image

File: assets/responsibleai/environments/responsibleai-text-ubuntu20.04-py38-cpu/context/Dockerfile [hunk 2]
```diff
@@ -23,6 +23,10 @@ RUN pip install 'azureml-dataset-runtime=={{latest-pypi-version}}' \
                 'azureml-telemetry=={{latest-pypi-version}}' \
                 'Werkzeug==2.2.3'
 
+# azureml-dataset-runtime[fuse] upper bound for pyarrow is 11.0.0
+# so we install pyarrow in extra step to avoid conflict
+RUN pip install 'pyarrow==14.0.1'
```
Comment (gaugup) [inline assets/responsibleai/environments/responsibleai-text-ubuntu20.04-py38-cpu/context/Dockerfile:28]: Why are we == pinning to 14.0.1? Looks like the conda file it was >=.

File: assets/responsibleai/environments/responsibleai-ubuntu20.04-py38-cpu/context/conda_dependencies.yaml (hunk 1 above)
Comment (gaugup) [reply assets/responsibleai/environments/responsibleai-ubuntu20.04-py38-cpu/context/conda_dependencies.yaml:13]: Sounds good.

File: assets/responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile [hunk 3]
```diff
@@ -36,6 +36,10 @@ RUN pip install 'shap==0.41.0' \
                 'scipy==1.10.0' \
                 'statsmodels==0.14.0'
 
+# azureml-dataset-runtime[fuse] upper bound for pyarrow is 11.0.0
+# so we install pyarrow in extra step to avoid conflict
+RUN pip install 'pyarrow==14.0.1'
```
Comment (gaugup) [inline assets/responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile:41]: Why are we == pinning to 14.0.1? Looks like the conda file it was >=.

File: assets/responsibleai/environments/responsibleai-text-ubuntu20.04-py38-cpu/context/Dockerfile (hunk 2 above)
Comment (hawestra) [reply assets/responsibleai/environments/responsibleai-text-ubuntu20.04-py38-cpu/context/Dockerfile:28]: https://msdata.visualstudio.com/Vienna/_git/AzureMlCli/pullrequest/1221644?_a=files

This is what azureml is doing which is why i pinned to that exact version

**Review by RubyZ10 — APPROVED** [review]

**Review by imatiach-msft — APPROVED** [review]

**Pull request merged**